src/components/app.jsx

```diff
@@ -1,9 +1,73 @@
-import React from "react";
-import { ReactComponent as Logo } from "../images/sun.svg";
+import React, { useState } from "react";
+import LocationInput from "./LocationInput";
+import WeatherDisplay from './WeatherDisplay';
+
 import "../App.css";
+import { fetchWeather } from "../utils";
 
 function App() {
-  return <h1>Weather</h1>;
+  const [weatherData, setWeather] = useState([{}]);
+  const [cityName, setCity] = useState('');
+  const [degreeMetric, setMetric] = useState('C');
+  const [timeframe, setTimeframe] = useState(1); // number of days
+  const [error, setError] = useState();
+
+  const onInputChange = (func) => (event) => func(event.target.value);
```

Comment:
This is just a higher order function. I chose this way because it allows you to have multiple functions operate on the same piece of data while keeping the code clean and portable. 

I called the first function when passing it down as [props](https://github.com/contently/weather-app/blob/912bd3fb68cbb481f19e8f98214ee036d21d45f2/src/components/app.jsx#L51) but if I had multiple components that might change the city name for instance, I could do `const updateCity = onInputChange(setCity);` and pass updateCity to every component that needed it. If `onInputChange` was a single function with two params, I would need to pass both `onInputChange` and `setCity` to every component it was needed instead of just `updateCity`.